### ISSUE TITLE
Do not allow the user to switch radios while pressing the PTT

### ISSUE THREAD

**Issue by Wetmelon** [ciribob/DCS-SimpleRadioStandalone#272]
The current behaviour of SRS is as follows:
- User presses PTT
   - SRS Broadcasts on radio 1 
- User presses radio 2
   - SRS stops broadcasting on radio 1
   - SRS switches to radio 2
   - SRS starts broadcasting on radio 2

This is *generally* ok, but it becomes less ideal when radio 1 and 2 only differ by a modifier.  I discovered that with the following settings:

![image](https://user-images.githubusercontent.com/3612694/44998756-0b02a000-af86-11e8-8fc0-5767c1b7e4d1.png)

And "Radio switch works as PTT" setting, it becomes very easy to accidentally broadcast on the wrong radio when stopping a transmission. I find that I press the modifier first then the broadcast button reliably but it's very easy to sequence them the wrong way when ending the transmission, so sometimes I let go of the modifer which immediately switches me to radio 1 and starts broadcasting.

I propose that switching radios should not be allowed while PTT is depressed.  This means the above example would look like:

- User presses PTT
   - SRS broadcasts on radio 1
- User presses Radio 2
   - SRS ignores input
- User releases PTT
   - SRS stops broadcasting on radio 1
- User presses radio 2
   - SRS switches to radio 2
- User presses PTT
   - SRS starts broadcasting on radio 2

This should work the same way if "Radio switch as PTT" is selected:

- User presses radio 1
   - SRS switches to radio 1 and broadcasts on radio 1
- User presses radio 2
   - SRS ignores input
- User releases radio 1
   - SRS stops broadcasting on radio 1
   - Note that SRS does NOT start broadcasting on radio 2 immediately
- User releases radio 2
   - SRS does nothing
- User presses radio 2
   - SRS switches to radio 2 and broadcasts on radio 2

**Comment by syn111** [comment]
This is not how real PTT behaves

**Comment by justin-lovell** [comment]
@syn111, I agree. However, SRS does not do concurrent transmissions over other radios when more than one PTT is used. In the real world, the buttons are gate-guarded or both radios would transmit concurrently.

The PR mentioning this issue would implement the logic as proposed by @Wetmelon.

However, it should be noted that the modifier on the same PTT key needs further work.
